jetbrains-core/src/software/aws/toolkits/jetbrains/core/executables/ExecutableCommon.kt

```diff
@@ -65,6 +65,34 @@ class ExecutableCommon {
             }
         }
 
+        @JvmStatic
+        fun checkSemVerVersionForParallelValidVersions(
+            version: SemVer,
+            minVersionLower: SemVer,
+            maxVersionLower: SemVer,
+            minVersionHigher: SemVer,
+            maxVersionHigher: SemVer,
```

Comment:
Thoughts on taking a List<Pair<SemVer, SemVer>> supportedRanges?

We can then add a new method that is just a single one and creates a list out of it
`fun checkSemVerVersionForParallelValidVersions(version:SemVer, lower:SemVer, higher:SemVer, executableName: String) = checkSemVerVersionForParallelValidVersions(version, listOf(lower to higher), executableName)`

Reply:
Pair sounds like a good idea, it'll be much cleaner and less confusing.
Since there's just 2 pairs, a list can be avoided, would improve readability?

Reply:
I think the list can simplify the code since we can loop and avoid so many if's and it can scale (let's hope we dont need >2...) but the point still stands

Reply:
Yep, makes sense, made the change!